File: infrastructure/lib/infrastructure-js-stack.ts
import {
    aws_dynamodb as dynamodb,
    aws_lambda as lambda,
    aws_logs as logs,
    Duration,
    Fn,
    Stack,
    StackProps
} from 'aws-cdk-lib';
import {Construct} from 'constructs';
import * as path from "path";

export class InfrastructureJsStack extends Stack {
    constructor(scope: Construct, id: string, props?: StackProps) {
        super(scope, id, props);

        const productsTable = dynamodb.Table.fromTableArn(this, 'dynamoTable', Fn.importValue('Products-JS-ExampleTableArn'));
        const lambdaJS = new lambda.Function(this, 'lambdaJSArm64', {
            description: "Kotlin Lambda JS Example",
            runtime: lambda.Runtime.NODEJS_16_X,
            architecture: lambda.Architecture.ARM_64,
            timeout: Duration.seconds(120),
            memorySize: 512,
            handler: 'kotlin-lambda-example-js-products.handleRequest',
            code: lambda.Code.fromAsset(path.join(__dirname, '../../build/dist/function.zip')),
            environment: {
                REGION: Stack.of(this).region,
            },
            logRetention: logs.RetentionDays.ONE_DAY
        });

        productsTable.grantReadData(lambdaJS);
    }
}

File: infrastructure/src/main/kotlin/nl.vintik.sample.infra/InfrastructureJSStack.kt
import software.amazon.awscdk.Duration
import software.amazon.awscdk.Fn
import software.amazon.awscdk.Stack
import software.amazon.awscdk.StackProps
import software.amazon.awscdk.services.dynamodb.Table
import software.amazon.awscdk.services.lambda.Code
import software.amazon.awscdk.services.lambda.Function
import software.amazon.awscdk.services.lambda.Runtime
import software.amazon.awscdk.services.logs.RetentionDays
import software.constructs.Construct

class InfrastructureJsStack(scope: Construct, id: String, props: StackProps) : Stack(scope, id, props) {
    init {
        val productsTable = Table.fromTableArn(this, "dynamoTable", Fn.importValue("Products-JS-ExampleTableArn"))
        val function = Function.Builder.create(this, "lambdaJS")
            .description("Kotlin Lambda JS Example")
            .handler("kotlin-lambda-example-js-products.handleRequest")
            .runtime(Runtime.NODEJS_18_X)
            .code(Code.fromAsset("../build/dist/function.zip"))
            .environment(
                mapOf("REGION" to of(this).region)
            )
            .logRetention(RetentionDays.ONE_WEEK)
            .memorySize(512)
            .timeout(Duration.seconds(120))
            .build()
        productsTable.grantReadData(function)
    }
}